api/node/__test__/js_value_conversion.spec.ts

```diff
@@ -310,7 +311,79 @@ test('get/set brush properties', (t) => {
   if (t.true((linearGradient instanceof SlintBrush))) {
     t.is((linearGradient as SlintBrush).toString(),
       "linear-gradient(90deg, rgba(63, 135, 166, 255) 0%, rgba(235, 248, 225, 255) 50%, rgba(246, 157, 60, 255) 100%)");
-  }
+  };
+
+  t.throws(() => {
+    instance.setProperty("ref-color", { red: "abc", blue: 0, green: 0, alpha: 0} );
+  },
+    {
+      code: 'NumberExpected',
+      message: 'Failed to convert napi value String into rust type `f64`',
+    }
+  );
+
+  t.throws(() => {
+    instance.setProperty("ref-color", { red: 0, blue: true, green: 0, alpha: 0} );
+  },
+    {
+      code: 'NumberExpected',
+      message: 'Failed to convert napi value Boolean into rust type `f64`',
+    }
+  );
+
+  t.throws(() => {
+    instance.setProperty("ref-color", { red: 0, blue: 0, green: true, alpha: 0} );
+  },
+    {
+      code: 'NumberExpected',
+      message: 'Failed to convert napi value Boolean into rust type `f64`',
+    }
+  );
+
+  t.throws(() => {
+    instance.setProperty("ref-color", { red: 0, blue: 0, green: 0, alpha: new SlintRgbaColor()} );
+  },
+    {
+      code: 'NumberExpected',
+      message: 'Failed to convert napi value Object into rust type `f64`',
+    }
+  );
+
+  t.throws(() => {
+    instance.setProperty("ref-color", { blue: 0, green: 0, alpha: 0} );
+  },
+    {
+      code: 'GenericFailure',
+      message: 'Property red is missing',
+    }
+  );
+
+  t.throws(() => {
+    instance.setProperty("ref-color", { red: 0, green: 0, alpha: 0} );
+  },
+    {
+      code: 'GenericFailure',
+      message: 'Property blue is missing',
+    }
+  );
+
+  t.throws(() => {
+    instance.setProperty("ref-color", { red: 0, blue: 0, alpha: 0} );
+  },
+    {
+      code: 'GenericFailure',
+      message: 'Property green is missing',
+    }
+  );
+
+  t.throws(() => {
+    instance.setProperty("ref-color", { red: 0, green: 0, blue: 0 } );
+  },
+    {
+      code: 'GenericFailure',
+      message: 'Property alpha is missing',
+    }
+  );
```

Comment:
I think we should make `alpha` an optional property, so that one can feed objects with just rgb. If the property is absent, it makes sense to assume that the color is meant to be opaque.

Reply:
Make sense!

Reply:
done